flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/common/utils/AvroToBigQuerySchemaTransform.java

```diff
@@ -0,0 +1,456 @@
+     * BigQuery equivalent, and constructs a BigQuery schema with the resulting fields.
+     *
+     * <p>For each Avro field, the method extracts the field's name and documentation (if
+     * available), and uses this information to build the corresponding BigQuery field.
+     *
+     * <p>The Avro schema must be of type Avro RECORD
+     *
+     * @param avroSchema The Avro schema to convert.
+     * @return The converted BigQuery {@link Schema}.
+     */
+    public static Schema getBigQuerySchema(org.apache.avro.Schema avroSchema) {
+        Preconditions.checkNotNull(avroSchema, "The avro schema of the record cannot be null.");
+        Preconditions.checkNotNull(
+                avroSchema.getType(), "The avro schema of the record must have a type.");
+        if (avroSchema.getType() != org.apache.avro.Schema.Type.RECORD) {
+            throw new IllegalArgumentException(
+                    "The Avro Schema must be of RECORD type to convert to BigQuery columns. Found Schema: "
+                            + avroSchema);
+        }
+        Preconditions.checkState(
+                !avroSchema.getFields().isEmpty(),
+                "The avro record schema must have at least one field.");
+        // Iterate over each record field and add them to the BigQuery schema.
+        List<Field> fields =
+                avroSchema.getFields().stream()
+                        .map(
+                                avroField -> {
+                                    Preconditions.checkNotNull(
+                                            avroField.name(),
+                                            "The avro field must have a name attribute.");
+                                    Field.Builder bigQueryFieldBuilder =
+                                            convertAvroFieldToBigQueryField(
+                                                            avroField.schema(), avroField.name(), 0)
+                                                    .toBuilder();
+                                    bigQueryFieldBuilder.setName(avroField.name());
+                                    if (avroField.doc() != null) {
+                                        bigQueryFieldBuilder.setDescription(avroField.doc());
+                                    }
+                                    return bigQueryFieldBuilder.build();
+                                })
+                        .collect(Collectors.toList());
+        return Schema.of(fields);
+    }
+
+    /**
+     * Converts an Avro field to a BigQuery field.
+     *
+     * @param avroField the Avro field to convert
+     * @param name the name of the field
+     * @param nestedLevel the current nesting level of the field
+     */
+    private static Field convertAvroFieldToBigQueryField(
+            org.apache.avro.Schema avroField, String name, int nestedLevel) {
+        if (nestedLevel > MAX_NESTED_LEVEL) {
+            throw new UnsupportedOperationException(
```

Comment:
Lets stick with IllegalArgumentException.
Also, in the error message your `nestedLevel` will always be 16 if error is being caught. Furthermore, `%s` for integer is incorrect format. Change error message to:
```
Field nesting in expected output's schema exceeds 15. BigQuery allows nesting up to a depth of 15.
```

Reply:
Thanks for pointing this out. Have done the required changes.